feat: add accounts count to delta and total stats

yaboiishere opened the pull request:
This PR includes a long-ish migration around ~12 hours locally 
resolves: #2040

yaboiishere committed:
feat: add accounts count to delta and total stats

yaboiishere committed:
fix: remove key_baundary

yaboiishere committed:
fix: add clause to skip tops delta_stat

yaboiishere committed:
fix: fix the way the delta stats are synced

yaboiishere committed:
fix: swap prev and current accounts in wrong calculation

yaboiishere committed:
chore: make dialyzer happy and remove IO prints

yaboiishere committed:
fix: ignore dialyzer in migration and fix typos in stats mutation

yaboiishere committed:
chore: refactor and update stats integration tests

yaboiishere committed:
chore: make function private to fix linting

sborrazas commented on lib/ae_mdw/db/mutations/stats_mutation.ex:
```diff
@@ -252,4 +262,35 @@ defmodule AeMdw.Db.StatsMutation do
+
+  defp get_accounts(state, from_txi, next_txi) do
+    state
+    |> Collection.stream(Model.Tx, {from_txi, next_txi})
```
I don't think this is right. `Model.Tx` keys are just txis, how is this key going to compare with a plain number?

yaboiishere replied:
Oh, it is supposed to be the other clause of stream that accepts a key_boundary fixing it

sborrazas commented on lib/ae_mdw/db/mutations/stats_mutation.ex:
```diff
@@ -252,4 +262,35 @@ defmodule AeMdw.Db.StatsMutation do
+  defp get_accounts(state, from_txi, next_txi) do
+    state
+    |> Collection.stream(Model.Tx, {from_txi, next_txi})
+    |> Stream.take_while(&match?(txi when txi < next_txi, &1))
```
No need for matching here `Stream.take_while(&(&1 < next_txi))`

yaboiishere committed:
fix: use the correct scope when checking for new accounts

yaboiishere committed:
Merge branch 'master' into account_stats

venimus committed:
Merge branch 'master' into account_stats

venimus committed:
feat: update account statistics handling and tests
- Refactor StatsMutation to use State.count_keys for account tracking
- Remove outdated migration for accounts in stats
- Add new migration to efficiently calculate account totals
- Enhance tests to verify account statistics increment logic
- Ensure API responses include accounts field in stats
